selfdrive/car/subaru/carcontroller.py

```diff
@@ -45,6 +51,49 @@ def update(self, CC, CS):
 
       self.apply_steer_last = apply_steer
 
+    # *** stop and go ***
+
+    throttle_cmd = False
+
+    if CS.CP.carFingerprint in PREGLOBAL_CARS:
+      if (c.enabled                                            # ACC active
+          and CS.car_follow == 1                             # lead car
+          and CS.out.standstill                              # must be standing still
+          and CS.close_distance > 3                          # acc resume min trigger threshold (m)
+          and CS.close_distance < 4.5                        # acc resume max trigger threshold (m)
+          and CS.close_distance > self.prev_close_distance): # distance with lead car is increasing
+        self.sng_acc_resume = True
+    elif CS.CP.carFingerprint not in PREGLOBAL_CARS:
+      # Record manual hold set while in standstill and no car in front
+      if CS.out.standstill and self.prev_cruise_state == 1 and CS.cruise_state == 3 and CS.car_follow == 0:
+        self.manual_hold = True
+      # Cancel manual hold when car starts moving
+      if not CS.out.standstill:
+        self.manual_hold = False
+      if (c.enabled                                            # ACC active
+          and not self.manual_hold
+          and CS.car_follow == 1                             # lead car
+          and CS.cruise_state == 3                           # ACC HOLD (only with EPB)
+          and CS.out.standstill                              # must be standing still
+          and CS.close_distance > 3                          # acc resume min trigger threshold (m)
+          and CS.close_distance < 4.5                        # acc resume max trigger threshold (m)
+          and CS.close_distance > self.prev_close_distance): # distance with lead car is increasing
+        self.sng_acc_resume = True
+      self.prev_cruise_state = CS.cruise_state
+
+    if self.sng_acc_resume:
+      if self.sng_acc_resume_cnt < 5:
+        throttle_cmd = True
+        self.sng_acc_resume_cnt += 1
+      else:
+        self.sng_acc_resume = False
+        self.sng_acc_resume_cnt = -1
```

Comment:
Any reason to only send for a certain number of frames?

Reply:
sending one frame did not trigger resume so I increased the frame count until it worked

Reply:
You can use `CC.cruiseControl.resume` as mentioned above